### PR TITLE
Blaze: Synchronize target devices

### PR DESCRIPTION
**Pull request by itsmeichigo**
Part of #11556

## Description

This PR updates `BlazeStore` and `BlazeAction` to support syncing target devices. Since the API is not ready, `BlazeStore` supports returning stubbed responses except for unit tests.

## Testing instructions

Just CI passing is sufficient.

## Screenshots

n/a

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by itsmeichigo** [comment]
🗒️  Following the discussion in peeHDf-2fw-p2#comment-1619, I'll have to add locales to the models and update Yosemite accordingly. This implementation will be updated with locale in the next PR.

File: Yosemite/Yosemite/Stores/BlazeStore.swift [hunk 1]
```diff
@@ -120,3 +122,61 @@ private extension BlazeStore {
...
+            } catch {
+                onCompletion(.failure(error))
+            }
+        }
+    }
+
+    /// Updates or Inserts specified BlazeTargetDevice Entities in a background thread
+    /// `onCompletion` will be called on the main thread.
+    ///
+    func upsertStoredTargetDeviceInBackground(readonlyDevices: [Networking.BlazeTargetDevice],
```
Comment (selanthiraiyan) [inline Yosemite/Yosemite/Stores/BlazeStore.swift]: Super nit
```suggestion
    func upsertStoredTargetDevicesInBackground(readonlyDevices: [Networking.BlazeTargetDevice],
```

File: Yosemite/Yosemite/Stores/BlazeStore.swift [hunk 2]
```diff
@@ -120,3 +122,61 @@ private extension BlazeStore {
...
+                                              onCompletion: @escaping () -> Void) {
+        let derivedStorage = sharedDerivedStorage
+        derivedStorage.perform { [weak self] in
```
Comment (selanthiraiyan) [inline Yosemite/Yosemite/Stores/BlazeStore.swift]: `self` can be removed as it is not used.

File: Yosemite/Yosemite/Stores/BlazeStore.swift [hunk 3]
```diff
@@ -120,3 +122,61 @@ private extension BlazeStore {
...
+                upsertStoredTargetDeviceInBackground(readonlyDevices: devices) {
+                    onCompletion(.success(devices))
+                }
+            } catch {
+                onCompletion(.failure(error))
+            }
+        }
+    }
+
+    /// Updates or Inserts specified BlazeTargetDevice Entities in a background thread
```
Comment (selanthiraiyan) [inline Yosemite/Yosemite/Stores/BlazeStore.swift]: "Updates or Inserts" - Should we remove the word "Updates" as we are always deleting any existing items?

File: Yosemite/Yosemite/Stores/BlazeStore.swift [hunk 4]
```diff
@@ -120,3 +122,61 @@ private extension BlazeStore {
...
+        NSClassFromString("XCTestCase") != nil
+    }
+
+    func mockResponse<T>(stubbedResult: T, onExecution: () async throws -> T) async throws -> T {
+        // skips stubbed result for unit tests
+        guard Self.isRunningTests else {
+            return stubbedResult
+        }
+        return try await onExecution()
+    }
```
Comment (selanthiraiyan) [inline Yosemite/Yosemite/Stores/BlazeStore.swift:181]: Nice idea to mock response.

**Review by selanthiraiyan — APPROVED** [review]
LGTM :shipit:

File: Yosemite/Yosemite/Stores/BlazeStore.swift (hunk 1 above)
Comment (itsmeichigo) [reply Yosemite/Yosemite/Stores/BlazeStore.swift]: Updated in af8228be109a34fa700db2b6208ce486c219e189.

File: Yosemite/Yosemite/Stores/BlazeStore.swift (hunk 2 above)
Comment (itsmeichigo) [reply Yosemite/Yosemite/Stores/BlazeStore.swift]: Also updated in af8228be109a34fa700db2b6208ce486c219e189.

File: Yosemite/Yosemite/Stores/BlazeStore.swift (hunk 3 above)
Comment (itsmeichigo) [reply Yosemite/Yosemite/Stores/BlazeStore.swift]: Also updated in af8228be109a34fa700db2b6208ce486c219e189.

**Pull request merged**